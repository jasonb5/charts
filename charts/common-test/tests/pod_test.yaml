suite: test pod
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should add extraContainers
    set:
      workload: deployment

      extraContainers:
        db:
          image:
            repository: mariadb
            tag: latest
    asserts:
      - matchSnapshot:
          path: spec.template.spec.containers[1]
        documentIndex: 0
      - hasDocuments:
          count: 2
  - it: should add initContainers
    set:
      workload: deployment

      initContainers:
        chmod:
          image:
            repository: busybox
            tag: latest

          args: ["chmod", "+0600"]
    asserts:
      - matchSnapshot:
          path: spec.template.spec.initContainers[0]
        documentIndex: 0
      - hasDocuments:
          count: 2
  - it: should overwrite defaults
    set:
      workload: deployment

      activeDeadlineSeconds: 20
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorterms:
            - matchFields:
              - key: kubernetes.io/hostname
                operator: "In"
                values:
                - node01
      dnsConfig:
        options:
        - name: ndots
          value: "1"
      dnsPolicy: ClusterFirst
      hostAliases:
      - hostnames:
        - backend
        ip: 192.168.0.1
      hostIPC: true
      hostNetwork: true
      hostPID: true
      hostUsers: true
      hostname: backend
      image:
        pullSecrets:
          docker.io:
            username: user1
            password: abcd1234
            email: user1@domain.io
      nodeName: node01
      nodeSelector:
        kubernetes.io/hostname: node01
      preemptionPolicy: Never
      priority: 20
      priorityClassName: system-node-critical
      podRestartPolicy: Always
      runtimeClassName: nvidia
      podSecurityContext:
        runAsUser: 65536
        runAsGroup: 65536
      serviceAccount:
        name: test
      terminationGracePeriodSeconds: 20
      tolerations:
      - effect: NoSchedule
        key: kubernetes.io/hostname
        operator: "Equal"
        value: node01
    asserts:
      - matchSnapshot: {}
        documentIndex: 1
      - matchSnapshot:
          path: spec.template.spec
        documentIndex: 0
      - hasDocuments:
          count: 3
  - it: should pass
    set:
      workload: deployment
    asserts:
      - matchSnapshot:
          path: spec.template.spec
        documentIndex: 0
      - hasDocuments:
          count: 2
