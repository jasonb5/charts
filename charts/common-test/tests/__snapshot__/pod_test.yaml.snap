should add extraContainers:
  1: |
    image: mariadb:latest
    name: db
should add initContainers:
  1: |
    args:
      - chmod
      - "+0600"
    image: busybox:latest
    name: chmod
should overwrite defaults:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        common/workload: default
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      ports:
        - name: default
          port: null
          protocol: TCP
          targetPort: null
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
        common/workload: default
      type: ClusterIP
  2: |
    activeDeadlineSeconds: 20
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorterms:
            - matchFields:
                - key: kubernetes.io/hostname
                  operator: In
                  values:
                    - node01
    containers:
      - image: busybox:0.1.0
        livenessProbe:
          tcpSocket:
            port: default
        name: common-test
        ports:
          - containerPort: null
            name: default
            protocol: TCP
        readinessProbe:
          tcpSocket:
            port: default
    dnsConfig:
      options:
        - name: ndots
          value: "1"
    dnsPolicy: ClusterFirst
    hostAliases:
      - hostnames:
          - backend
        ip: 192.168.0.1
    hostIPC: true
    hostNetwork: true
    hostPID: true
    hostUsers: true
    hostname: backend
    imagePullSecrets:
      - name: RELEASE-NAME-common-test-regcreds
    nodeName: node01
    nodeSelector:
      kubernetes.io/hostname: node01
    preemptionPolicy: Never
    priority: 20
    priorityClassName: system-node-critical
    restartPolicy: Always
    runtimeClassName: nvidia
    securityContext:
      runAsGroup: 65536
      runAsUser: 65536
    serviceAccountName: test
    terminationGracePeriodSeconds: 20
    tolerations:
      - effect: NoSchedule
        key: kubernetes.io/hostname
        operator: Equal
        value: node01
should pass:
  1: |
    containers:
      - image: busybox:0.1.0
        livenessProbe:
          tcpSocket:
            port: default
        name: common-test
        ports:
          - containerPort: null
            name: default
            protocol: TCP
        readinessProbe:
          tcpSocket:
            port: default
